main.tf

```diff
@@ -0,0 +1,178 @@
+  arn       = var.sns_topic_arn != "" ? var.sns_topic_arn : aws_sns_topic.this[0].arn
+
+  dynamic "input_transformer" {
+    for_each = each.key == "ECSTaskStateChange" ? [1] : []
+    content {
+      input_paths = {
+        "account"              = "$.account"
+        "detail"               = "$.detail"
+        "detail-type"          = "$.detail-type"
+        "id"                   = "$.id"
+        "region"               = "$.region"
+        "resources"            = "$.resources"
+        "time"                 = "$.time"
+        "taskDefinitionArn"    = "$.detail.taskDefinitionArn"
+        "lastStatus"           = "$.detail.lastStatus"
+        "desiredStatus"        = "$.detail.desiredStatus"
+        "containerInstanceArn" = "$.detail.containerInstanceArn"
+        "healthStatus"         = "$.detail.healthStatus"
+        "stopCode"             = "$.detail.stopCode"
+        "stoppedReason"        = "$.detail.stoppedReason"
+        "containers"           = "$.detail.containers"
+      }
+      input_template = <<EOF
+{
+  "version": "1.0",
+  "source": "custom",
+  "id": <id>,
+  "content": {
+    "textType": "client-markdown",
+    "title": <detail-type>,
+    "description": "*Event Detail:* \r\n• Task Definition: <taskDefinitionArn>\r\n• Last: <lastStatus>\r\n• Desired: <desiredStatus> \r\n• account: <account> \r\n• time: <time>",
+    "keywords": [<region>]
+  },
+  "metadata": {
+    "threadId": <id>,
+    "summary": <detail-type>,
+    "eventType": <detail-type>,
+    "relatedResources": <resources>,
+    "additionalContext": {
+      "account": <account>,
+      "time": <time>
+    }
+  }
+}
+EOF
+    }
+  }
+
+  dynamic "input_transformer" {
+    for_each = each.key == "ECSDeploymentStateChange" ? [1] : []
+    content {
+      input_paths = {
+        "account"      = "$.account"
+        "detail"       = "$.detail"
+        "detail-type"  = "$.detail-type"
+        "id"           = "$.id"
+        "region"       = "$.region"
+        "resources"    = "$.resources"
+        "time"         = "$.time"
+        "eventType"    = "$.detail.eventType"
+        "eventName"    = "$.detail.eventName"
+        "deploymentId" = "$.detail.deploymentId"
+        "reason"       = "$.detail.reason"
+      }
+      input_template = <<EOF
+{
+  "version": "1.0",
+  "source": "custom",
+  "id": <id>,
+  "content": {
+    "textType": "client-markdown",
+    "title": <detail-type>,
+    "description": "*Event Detail:* \r\n• <eventType> - <eventName>\r\n• Deployment: <deploymentId>\r\n• Reason: <reason> \r\n• account: <account> \r\n• time: <time>",
+    "keywords": [<region>]
+  },
+  "metadata": {
+    "threadId": <id>,
+    "summary": <detail-type>,
+    "eventType": <detail-type>,
+    "relatedResources": <resources>,
+    "additionalContext": {
+      "account": <account>,
+      "time": <time>
+    }
+  }
+}
+EOF
+    }
+  }
+
+  dynamic "input_transformer" {
+    for_each = each.key == "ECSServiceAction" ? [1] : []
```

Comment:
For the next iteration, we need to think about how to handle non-ECS events and custom events. Most probably we will need to have a default transformer and custom transformers per custom rule